.github/workflows/tests.yaml

```diff
@@ -0,0 +1,40 @@
+name: Tests build
+
+on: [push, pull_request]
+
+env:
+  DVC_TEST: "true"
+  HOMEBREW_NO_AUTO_UPDATE: 1
+  SHELL: /bin/bash
+
+jobs:
+  tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, windows-2019, macos-10.15]
+        pyv: [3.6, 3.7, 3.8]
+        exclude:
+          - os: windows-2019
+            pyv: 3.8
+    steps:
+    - uses: actions/checkout@v2.3.2
+    - name: Set up Python
+      uses: actions/setup-python@v2.1.2
+      with:
+        python-version: ${{ matrix.pyv }}
+    - name: install
+      run: pip install ".[all,tests]"
+    - name: setup git
+      run: |
+        git config --global user.email "dvctester@example.com"
+        git config --global user.name "DVC Tester"
+    - name: run tests
+      run: python -m tests -n=2
+    - name: run codecov
+      if: github.event.action == 'push'
+      run: |
+        pip install codecov
+        codecov
```

Comment:
@efiop could you set `CODECOV_TOKEN`?

Reply:
Added to master, should be set.

Reply:
@skshetry pointed out that this is not the best idea to add it publically (which I did), so now I've added a CODECOV_TOKEN secret. Should be usable now.

Reply:
@pared, [`codecov-action`](https://github.com/codecov/codecov-action) supports tokenless uploads using Github Actions, only caveat being we need to change report format to xml: https://github.com/codecov/codecov-action#example-workflowyml-with-codecov-action

Reply:
@efiop , sorry I meant to set it as repository secret, should have explained that